core/templates/pages/contributor-dashboard-page/services/contribution-opportunities-backend-api.service.ts

```diff
@@ -215,6 +219,28 @@ export class ContributionOpportunitiesBackendApiService {
       return [];
     }
   }
+
+  async saveUserLanguagePreferenceAsync(languageCode: string):
+  Promise<Object> {
+    const postData: {'language_code': string} = {
+      language_code: languageCode,
+    };
+    const body = new FormData();
+    body.append('payload', JSON.stringify(postData));
+    const response = this.http.post(
+      '/preferredtranslationlanguage', body).toPromise();
+    return response;
```

Comment:
The post call doesn't seem to return anything, so I think we can return void by specifying `<void>`:

```suggestion
    return this.http.post<void>(
      '/preferredtranslationlanguage', body).toPromise();
```

Reply:
Actually, I am facing issues with this. We need to return something in the post call. Otherwise, we will get the following error during `self.post_json()`:
![image](https://user-images.githubusercontent.com/75239620/169136137-9ef9d2a3-21ff-409c-bb3e-9dffe924e48a.png)


To avoid this, I have added `self.render_json({})` in the post call. We are doing this at other locations as well.
For example:
https://github.com/oppia/oppia/blob/0af67f015ab36e2cb2fc5efbeda157a4528a6a1d/core/controllers/profile.py#L537-L560

Now, what we should do over here? Is there any other alternative way to deal with this issue?